llvm/include/llvm/RemoteCachingService/Client.h

```diff
@@ -221,25 +225,107 @@ class CASDBClient {
     virtual void
     saveFileAsyncImpl(std::string FilePath,
                       std::shared_ptr<AsyncCallerContext> CallCtx) = 0;
-    virtual Expected<Response> receiveNextImpl() = 0;
+    virtual llvm::Expected<Response> receiveNextImpl() = 0;
+  };
+
+  class GetAsyncQueue : public AsyncQueueBase {
+    virtual void anchor() override;
+
+  public:
+    virtual ~GetAsyncQueue() = default;
+
+    void getAsync(std::string CASID,
+                  llvm::Optional<std::string> OutFilePath = llvm::None,
```

Comment:
I think we need those since the GRPC protocol support that. We also probably need to implementing returning something larger than 4MB in a file to workaround GRPC limit (unless we switch to streaming API).